src/power_grid_model_io_native/_core/error_handling.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+"""
+Error handling
+"""
+
+from power_grid_model._core.error_handling import (
+    _ERROR_MESSAGE_PATTERNS,
+    PGM_NO_ERROR,
+    PGM_REGULAR_ERROR,
+    PGM_SERIALIZATION_ERROR,
+    VALIDATOR_MSG,
+    PowerGridBatchError,
+    PowerGridError,
+    PowerGridSerializationError,
+)
+
+from power_grid_model_io_native._core.power_grid_model_io_core import pgm_io_core as pgmic
+
+
+def _interpret_error(message: str, decode_error: bool = True) -> PowerGridError:
+    if decode_error:
+        for pattern, type_ in _ERROR_MESSAGE_PATTERNS.items():
+            if pattern.search(message) is not None:
+                return type_(message)
+
+    return PowerGridError(message)
```

Comment:
you can also get this from the PGM instead - then, you don't need to load the `_ERROR_MESSAGE_PATTERNS` from there. You will need to add separate error message patterns + a decoder for the errors only specific to PGM-IO-Native, though, so please keep the decoder here as well

Reply:
used the _interpret_error from PGM, removed the _ERROR_MESSAGE_PATTERNS import. Kept a decoder as a comment for now